origocli/commands/status.py

```diff
@@ -30,53 +30,54 @@ def login(self):
 
     def default(self):
         self.log.info("StatusCommand.default()")
-        if self.arg("statusid"):
+        if self.arg("trace_id"):
             self.status_for_id()
         else:
             self.print("Invalid command")
 
     @staticmethod
-    def add_status_for_id_rows(out, statusid, statuses):
+    def add_status_for_id_rows(out, trace_id, trace_events):
         finished = False
-        run_status = statuses[-1]["run_status"]
-        status = statuses[-1]["status"]
-        for el in statuses:
-            if el["run_status"] == "FINISHED":
-                el["done"] = False
-                if el["status"] == "OK":
-                    el["done"] = True
+        trace_status = trace_events[-1]["trace_status"]
+        trace_event_status = trace_events[-1]["trace_event_status"]
+
+        for i, trace_event in enumerate(trace_events, 1):
+            if trace_event["trace_status"] == "FINISHED":
+                trace_event["done"] = False
+                if trace_event["trace_event_status"] == "OK":
+                    trace_event["done"] = True
                     finished = True
-                out.add_row(el)
+                    out.add_row(trace_event)
```

Comment:
Skal denne virkelig flyttes inn i den innerste `if`-testen?

Reply:
Sikkert lest det feil eller misforstått hva den prøver å oppnå, men som den var, legger den ikke til alle `trace_event_status != OK` som rader _inkl_ siste fra lista (fordi `finished=False`)?

```
+-------+----------------------+--------------+--------------------+
| Done  | Trace ID             | Trace status | Trace event status |
+-------+----------------------+--------------+--------------------+
| False | failed-trace-123-abc | FINISHED     | FAILED             |
| False | failed-trace-123-abc | FINISHED     | FAILED             |
| False | failed-trace-123-abc | FINISHED     | FAILED             |
+-------+----------------------+--------------+--------------------+
```

Reply:
Ok, er enig i at det er noe krøll i den gamle logikken der. Tanken er vel at den skal vise én oppsummert status (i motsetning til full history). 

Den prøver å finne den som har `trace_status` == FINISHED. Men, i den gamle logikken så satte den bare `finished = True` om `trace_event_status` == OK. Så om pipeline er FAILED så vil den vise **_to_** rader, både den som har FINISHED + FAILED og den som er sist i lista (inne i `if not finished:` blokka) - og de to er antakelig samme rad?? 🤔 

Hva med å lage en ny metode `find_latest_event` og gjøre noe ala:
```
latest_event = find_latest_event(trace_events) # finner den som er FINISHED, eller returnerer siste i lista
trace_status = latest_event["trace_status"]

out.add_row(
    {
        "done": trace_status == "FINISHED",
        "trace_id": trace_id,
        "trace_status": trace_status
        "trace_event_status": latest_event["trace_event_status"],
    }
)